### ISSUE TITLE
Dealing with additionalProperties

### ISSUE THREAD

**Issue by amarzavery** [Azure/azure-openapi-validator#134]
- `"additionalProperties" inside the model `"properties"` 
If a model contains a property named "additionalProperties" then issue a warning.

example:
```json5
{ 
 "Foo": {
    "properties": {
         "additionalProperties": {
            "type": "object"
         }
     }
  }
}
```

- **conflict**  `"additionalProperties" inside the model `"properties"` as well as outside the model `"properties"`.
If a model contains a property named additionalProperties and also contains additionalProperties on the outside (sibling of properties) then the linter should throw an error. Because having additionalProperties on the outside has a special meaning in Json schema.

example 1:
```json5
{ 
 "Foo": {
    "properties": {
         "additionalProperties": {
            "type": "object"
         }
     },
    "additionalProperties": true
  }
}
```
example2:
```json5
{ 
 "Foo": {
    "properties": {
         "additionalProperties": {
            "type": "object"
         }
     },
    "additionalProperties": {
       "type": "string" // could be "object" | "number" | any other valid value for the type
    }
  }
}
```
